src/pykeen/triples/triples_factory.py

```diff
@@ -121,24 +156,44 @@ def _map_triples_elements_to_ids(
-        logger.warning(
-            f"You're trying to map triples with {num_no_head + num_no_tail} entities and {num_no_relation} relations"
-            f" that are not in the training set. These triples will be excluded from the mapping.",
-        )
-        non_mappable_triples = head_filter | relation_filter | tail_filter
-        head_column = head_column[~non_mappable_triples, None]
-        relation_column = relation_column[~non_mappable_triples, None]
-        tail_column = tail_column[~non_mappable_triples, None]
-        logger.warning(
-            f"In total {non_mappable_triples.sum():.0f} from {triples.shape[0]:.0f} triples were filtered out",
-        )
-
-    triples_of_ids = np.concatenate([head_column, relation_column, tail_column], axis=1)
+    if not is_triples:
+        timestamp_filter = timestamp_column < 0
+        num_no_timestamp = timestamp_filter.sum()
+    
+    if is_triples:
+        if (num_no_head > 0) or (num_no_relation > 0) or (num_no_tail > 0):
+            logger.warning(
+                f"You're trying to map triples with {num_no_head + num_no_tail} entities and {num_no_relation} relations"
+                f" that are not in the training set. These triples will be excluded from the mapping.",
+            )
+            non_mappable_triples = head_filter | relation_filter | tail_filter
+            head_column = head_column[~non_mappable_triples, None]
+            relation_column = relation_column[~non_mappable_triples, None]
+            tail_column = tail_column[~non_mappable_triples, None]
+            logger.warning(
+                f"In total {non_mappable_triples.sum():.0f} from {triples.shape[0]:.0f} triples were filtered out",
+            )
+        triples_of_ids = np.concatenate([head_column, relation_column, tail_column], axis=1)
+    else:
+        if (num_no_head > 0) or (num_no_relation > 0) or (num_no_tail > 0) or (num_no_timestamp > 0):
+            logger.warning(
+                f"You're trying to map quadruples with {num_no_head + num_no_tail} entities, {num_no_relation} relations "
+                f"and {num_no_timestamp} timestamps "
+                f"that are not in the training set. These quadruples will be excluded from the mapping.",
+            )
+            non_mappable_quadruples = head_filter | relation_filter | tail_filter | timestamp_filter
+            head_column = head_column[~non_mappable_quadruples, None]
+            relation_column = relation_column[~non_mappable_quadruples, None]
+            tail_column = tail_column[~non_mappable_quadruples, None]
+            timestamp_column = timestamp_column[~non_mappable_quadruples, None]
+            logger.warning(
+                f"In total {non_mappable_quadruples.sum():.0f} from {triples.shape[0]:.0f} quadruples were filtered out",
+            )
+        triples_of_ids = np.concatenate([head_column, relation_column, tail_column, timestamp_column], axis=1)
```

Comment:
This part duplicates quite a bit of code. To reduce the duplication, I suggest to adapt the code similar to the following (*disclaimer*: written without running the code):
```python
id_columns = [head_column, relation_column, tail_column]
filter_columns = [
    ([COLUMN_HEAD, COLUMN_TAIL], "entities"),
    (COLUMN_RELATION, "relations"),
]
if timestamp_to_id is not None:
    timestamp_getter = np.vectorize(timestamp_to_id.get)
    timestamp_column = timestamp_getter(triples[:, 3:4], [-1])
    id_columns.append(timestamp_column)
    filter_columns.append((COLUMN_TIMESTAMP, "timestamps"))
array_of_ids = numpy.stack(id_columns, axis=1)

drop_mask = None
for columns, label in filter_columns:
    column_ids = array_of_ids[:, columns]
    invalid_ids = column_ids < 0
    num_invalid_ids = invalid_ids.sum()
    if num_invalid_ids:
        logger.warning(
            f"You're trying to map triples with {num_invalid_ids:_} {label} "
            "that are not in the training set. "
            "These triples will be excluded from the mapping.",
        )
        drop_mask = invalid_ids if drop_mask is None else drop_mask | ~invalid_ids
if drop_mask is not None:
    logger.warning(
        f"In total {drop_mask.sum():_} from {array_of_ids.shape[0]:_} "
        "triples/quadruples were filtered out",
    )
    array_of_ids = array_of_ids[~drop_mask]
unique_mapped_triples = np.unique(ar=triples_of_ids, axis=0)
return torch.tensor(unique_mapped_triples, dtype=torch.long)
```